code/datums/components/onwear_mood.dm

```diff
@@ -0,0 +1,50 @@
+// Add to clothing to give the wearer a mood buff and a unique examine str
+
+/datum/component/onwear_mood
+	/// the event the wearer experiences
+	var/datum/mood_event/saved_event
+	/// examine string added to examine
+	var/examine_string
+
+/datum/component/onwear_mood/Initialize(clear_after, datum/mood_event/saved_event, examine_string)
+
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	if(istype(saved_event))
```

Comment:
If there's no default for `saved_event`, and there doesn't appear to be, you don't need this check.

Reply:
Also this check as you have it right now will never pass with what you wrote, as you are giving it paths, and not instances. Make sure you test that this is working.